bundle/direct/dstate/state.go

```diff
@@ -113,7 +117,12 @@ func (db *DeploymentState) Open(path string) error {
 		return err
 	}
 
-	err = json.Unmarshal(data, &db.Data)
+	// Use json.Decoder with UseNumber() to preserve precision of large integers.
```

Comment:
This is a potentially good change but deserves its own PR + tests.

In general, let's keep each PR focussed on one issue (unless they are so tangled, they have to be solved together - that does not seem to be the case here).